source/apiVolontaria/apiVolontaria/serializers.py

```diff
@@ -1,72 +1,36 @@
 import re
 
 from rest_framework import serializers
+from rest_framework.compat import authenticate
 
 from django.contrib.auth.models import User
-from django.contrib.auth import authenticate
-
-from django.core import exceptions
+from django.utils.translation import ugettext_lazy as _
 
 from .models import ActivationToken, Profile
 
 
 class AuthCustomTokenSerializer(serializers.Serializer):
     login = serializers.CharField()
-
-    password = serializers.CharField()
-
-    def validate_email(self, email):
-        if len(email) > 6:
-            if re.match(
-                    '[A-Za-z0-1\.-]+'
-                    '@[A-Za-z0-1\.-]+'
-                    '\.[A-Za-z0-1]{2,4}',
-                    email
-            ):
-                return 1
-        return 0
+    password = serializers.CharField(style={'input_type': 'password'})
 
     def validate(self, attrs):
-        email_or_username = attrs.get('login')
+        login = attrs.get('login')
         password = attrs.get('password')
 
-        if email_or_username and password:
-            # Check if user sent email
-            if self.validate_email(email_or_username):
-                email_exist = User.objects.filter(
-                    email=email_or_username,
-                )
-
-                if not email_exist.count():
-                    # Email is not use
-                    msg = "This email doesn't have account"
-                    raise exceptions.ValidationError(msg)
-
-                email_or_username = email_exist[0].username
-            else:
-                user_exist = User.objects.filter(
-                    username=email_or_username
-                ).count()
-
-                if not user_exist:
-                    # User doesn't exist
-                    msg = "This username doesn't have account"
-                    raise exceptions.ValidationError(msg)
-
-            user = authenticate(
-                username=email_or_username,
-                password=password
-            )
+        if login and password:
+            user_obj = User.objects.filter(email=login)
+            if user_obj:
+                login = user_obj.username
```

Comment:
Filter can return multiple element if we have a bug of unique in the future, if you want to use `user_obj.username` directly without verification it's better to use `get()` in place of `filter()`.

```
user_obj = User.objects.get(email=login)
user_obj = User.objects.filter(email=login)
```

Reply:
That's what I initially did, but if no user is found, it will throw a User.DoesNotExist error. I wondered if it was better to avoid a try/catch by using filter(). 

I'll check how we can make the email field unique

Reply:
With this example you only have 5 lines in place of 3, it's not to bad :
```
try:
    user_obj = User.objects.get(email=login)
    login = user_obj.username
except:
    pass
```

I just want to be sure to not have a problem if one day we have two people with the same email. If our rules of authentication change witht the time, people will maybe forget to check this specific part of the code.